src/app/commands/cmd_merge_down_layer.cpp

```diff
@@ -81,85 +76,17 @@ void MergeDownLayerCommand::onExecute(Context* context)
 
   Tx tx(writer, friendlyName(), ModifyDocument);
 
-  for (frame_t frpos = 0; frpos<sprite->totalFrames(); ++frpos) {
-    // Get frames
-    Cel* src_cel = src_layer->cel(frpos);
-    Cel* dst_cel = dst_layer->cel(frpos);
-
-    // Get images
-    Image* src_image;
-    if (src_cel != NULL)
-      src_image = src_cel->image();
-    else
-      src_image = NULL;
-
-    ImageRef dst_image;
-    if (dst_cel)
-      dst_image = dst_cel->imageRef();
-
-    // With source image?
-    if (src_image) {
-      int t;
-      int opacity;
-      opacity = MUL_UN8(src_cel->opacity(), src_layer->opacity(), t);
-
-      // No destination image
-      if (!dst_image) {  // Only a transparent layer can have a null cel
-        // Copy this cel to the destination layer...
-
-        // Creating a copy of the image
-        dst_image.reset(
-          render::rasterize_with_cel_bounds(src_cel));
-
-        // Creating a copy of the cell
-        dst_cel = new Cel(frpos, dst_image);
-        dst_cel->setPosition(src_cel->x(), src_cel->y());
-        dst_cel->setOpacity(opacity);
-
-        tx(new cmd::AddCel(dst_layer, dst_cel));
-      }
-      // With destination
-      else {
-        gfx::Rect bounds;
-
-        // Merge down in the background layer
-        if (dst_layer->isBackground()) {
-          bounds = sprite->bounds();
-        }
-        // Merge down in a transparent layer
-        else {
-          bounds = src_cel->bounds().createUnion(dst_cel->bounds());
-        }
-
-        doc::color_t bgcolor = app_get_color_to_clear_layer(dst_layer);
-
-        ImageRef new_image(doc::crop_image(
-            dst_image.get(),
-            bounds.x-dst_cel->x(),
-            bounds.y-dst_cel->y(),
-            bounds.w, bounds.h, bgcolor));
-
-        // Draw src_cel on new_image
-        render::rasterize(
-          new_image.get(), src_cel,
-          -bounds.x, -bounds.y, false);
-
-        // First unlink the dst_cel
-        if (dst_cel->links())
-          tx(new cmd::UnlinkCel(dst_cel));
-
-        // Then modify the dst_cel
-        tx(new cmd::SetCelPosition(dst_cel,
-            bounds.x, bounds.y));
-
-        tx(new cmd::ReplaceImage(sprite,
-            dst_cel->imageRef(), new_image));
-      }
-    }
-  }
-
-  document->notifyLayerMergedDown(src_layer, dst_layer);
-  document->getApi(tx).removeLayer(src_layer); // src_layer is deleted inside removeLayer()
+  DocRange range;
+  range.selectLayer(writer.layer());
+  range.selectLayer(dst_layer);
+  const bool newBlend = Preferences::instance().experimental.newBlend();
+
+  tx(new cmd::FlattenLayers(sprite,
+    range.selectedLayers(),
+    int(newBlend) |
```

Comment:
I think both options are valid (we've used both in Aseprite).
1. In case you go with flags, you can use the new [LAF_ENUM_FLAGS()](https://github.com/aseprite/laf/commit/6abf784cd9a131c85cab016ddbc5fa22199c9211) macro to overload the `operator|`.
2. Otherwise, using bitfields, we'll need an `Options()` constructor (e.g. like [DocDiff](https://github.com/aseprite/aseprite/blob/ec42689b820cc025e685e4f2bc3f230a76bb1e89/src/app/doc_diff.h#L29))

Both options are fine :+1: